components/governance/org.wso2.carbon.governance.api/src/main/java/org/wso2/carbon/governance/api/generic/GenericArtifactManager.java

```diff
@@ -83,25 +87,35 @@ public GenericArtifactManager(Registry registry, String mediaType,
 
     /**
      * Constructor accepting an instance of the registry, and key identifying the type of manager.
-     * 
+     *
      * @param registry the instance of the registry.
      * @param key      the key short name of the artifact type.
+     * @throws RegistryException
      */
     public GenericArtifactManager(Registry registry, String key) throws RegistryException {
         try {
             GovernanceArtifactConfiguration configuration =
                     GovernanceUtils.findGovernanceArtifactConfiguration(key, registry);
-            this.artifactNameAttribute = configuration.getArtifactNameAttribute();
-            this.artifactNamespaceAttribute = configuration.getArtifactNamespaceAttribute();
-            this.artifactElementNamespace = configuration.getArtifactElementNamespace();
-            manager = new GovernanceArtifactManager(registry, configuration.getMediaType(),
-                    this.artifactNameAttribute, this.artifactNamespaceAttribute,
-                    configuration.getArtifactElementRoot(), this.artifactElementNamespace,
-                    configuration.getPathExpression(), configuration.getLifecycle(),
-                    configuration.getValidationAttributes(), configuration.getRelationshipDefinitions());
-            this.mediaType = configuration.getMediaType();
+            if (configuration != null) {
+                this.artifactNameAttribute = configuration.getArtifactNameAttribute();
```

Comment:
Thanks for the comment.

This fix is done to handle the error of trying to add an artifact when the artifact type is not added to the registry.

Use case : When someone attempt to add an artifact from a registry client, if the artifact type is not added before (not in the registry) this will handle that error.